RecoLocalTracker/SiPixelClusterizer/plugins/PixelThresholdClusterizer.cc

```diff
@@ -119,6 +144,11 @@ void PixelThresholdClusterizer::clusterizeDetUnitT( const T & input,
   
   detid_ = input.detId();
   
+  // Set separate cluster threshold for L1 (needed for phase1)
+  auto clusterThreshold = theClusterThreshold;
+  int layer = (DetId(detid_).subdetId()==1) ? PXBDetId(detid_).layer() : 0;
```

Comment:
This is strange. According to this page: https://github.com/cms-sw/cmssw/tree/master/Geometry/TrackerNumberingBuilder
in the old detector the layer number is encoded in the detid starting at bit 16 while in the phase 1 detector the layer number starts at bit 20. According to that it is not possible that you get the right layer number if you decode phase 1 detids with the PXBDetId class.